### ISSUE TITLE
Intelisense suggestion widget is too verbose with screen readers

### ISSUE THREAD

**Issue by pawelurbanski** [Microsoft/vscode#65499]
I would like to propose certain changes to screen reader mode behavior.
 
Right now the suggestion alert is very verbose to the point of
 interrupting the flow when coding. The message posted to the alert div contains:
 (Current item), suggestion, has details, item label

I know that those messages differ based on the fact if there are item details or not. The message read aloud by the screen reader can contain a summary or more details besides the function / object / property name etc.

Unfortunately, the screen reader is reading the "suggestion, has details" suffix and so on which is too detailed, and I guess very annoying when you get to hear it after every refresh of a suggestion list.
The desired behavior should be announcing just the current item having focus - as the basic behavior. The reason is that it allows for quick listening and completing code.
Another usability comes from the fact that Visual Studio Code is alerting the following string:
 Accepting 'current item' did insert ... and so on. This is a usability bug, for I get to hear it after hitting enter on every desireable selection... Once the item is inserted it is in the code window content and can / should be easily read / navigable with regular screen reader review functions - just like simple text.
I've spent some time with the suggestion widget code, and understand the design, but there are parts, which are not so easy to deal with such as the alert after accepting code suggestions.
What would be the best way to provide fixes or configuration settings what would control the verbosity level?

**Comment by pawelurbanski** [comment]
I started fixing the code in the suggestion widget module. My results are as follows:

### Only currently suggested items are being read - no suffixes.
The code needs some cleanup - I'm not a TypeScript developer by any means. We still need to deal with the case when someone needs to lookup the documentation.

### The documentation lookup case.
When you are coding using a screen reader, your focus on the very line / suggested item. It is impossible to see the sidebar of the screen or an overlay window at below the code.
I would rather have the option of using a shortcut to focus if the docs are available when I need it, and just be notified, if for some reason they are not.

### Still getting the "Accepting <Completion Item Name> did insert the following" message.
I initially thought, that it is fired when the item is accepted, just before the editor.focus() call.
I removed the alert(...) function, but the behavior is still the same.

**Comment by pawelurbanski** [comment]
I closed the issue by mistake. Would be glad if someone could reopen it, so we can keep on working on the fixes.

**Comment by jrieken** [comment]
@pawelurbanski I have merged your PR after a little clean-up. Thanks so far. I'd like to keep iterating on this, please let me know how it works now. This change will be an insiders version coming end of this week (once we have released 1.32)

**Comment by pawelurbanski** [comment]
@jrieken I will install the insiders build once it is released, and do the following:
1. Check how the fixes work out.
2. Finish and release the basic NVDA screen reader add-on.
3. Propose changes to the accessibility page on wiki with hits for optimal configuration for screen readers, and add a link to the add-on.

I do have another 2 specific ideas for fixes, but will start working on them once the insiders build is released. Using VS Code itself, with its tooling will make it much easier to provide quality fixes.
As I said before, I am ready and avilable to give a hand from time to time to you or the VS Code team or to the community.

Thank you for making those changes a part of VS Code.

**Comment by jrieken** [comment]
> As I said before, I am ready and avilable to give a hand from time to time to you or the VS Code team or to the community.

That is awesome ❤️

**Comment by pawelurbanski** [comment]
I’ve tested committed changes using the Insiders build 1.33.0 as you suggested.
There is only one issue left. I wonder if my changes were not included in the pull request...

The function in: vs/base/browser/ui/aria/aria.ts reports how many times a suggested item occured.
It is not useful at all, and interrupts the flow. Variables or functions will be repeated through the code, but the information how many times they showed up in the suggestion widget has no value.
We could compare it to placing hover tooltips with such information for sighted users...
The change is quite trivial – let me know if I should create another pull request after making those changes, or you will be able to fix it yourself.

The alert function should just insert and anounce the item without the counter value.

Greetings, Pawel

**Comment by jrieken** [comment]
@pawelurbanski Thanks for reporting back. Yeah, I did remove the change in the aria.ts because I couldn't relate it to the IntelliSense widget. However, I am OK to nuke it based on your feedback. Just checking with @bpasero who added this a while back via https://github.com/Microsoft/vscode/commit/9ace9db63cfaae96cf6217fc50690ab869f80223 if there was any strong reason for it?

**Comment by bpasero** [comment]
@jrieken as far as I can tell, a screen reader would not announce again if the same message is being put. With the trick of having a counter, we enforce reading a message again even if it was already read. Maybe a better fix would be to unset the message after a delay? Not sure if that would break things though...

**Comment by pawelurbanski** [comment]
Let me give you an example how it works on my and the screen reader side:
The code is for example:
Var notifyX = window.addEventListener(...);
... some more code...
Var notifyY = window.addEventListener(...);

When getting the intelisense suggestion for the notifyY part the altert says: addEventListener occured 2 times.
I had my local branch wich was notifying completion items correctly wiythout the counter.
I’m quite sure that it was committed with the other fixes in the pull request.

**Comment by jrieken** [comment]
I have extracted the aria#alert issue into #70288 so that we can close this. Removing the alert-repeat-message sounds reasonable to me but need more testing as it is used in more places.
